webui/react/src/pages/F_ExpList/F_ExperimentList.tsx

```diff
@@ -111,10 +163,23 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
     } finally {
       setIsLoading(false);
     }
-  }, [page, experimentFilters, canceler.signal]);
+  }, [page, experimentFilters, canceler.signal, sortString]);
 
   const { stopPolling } = usePolling(fetchExperiments, { rerunOnNewFn: true });
 
+  // TODO: poll?
+  useEffect(() => {
+    let mounted = true;
+    getProjectColumns({ projectId: project.id }).then((c) => {
+      if (mounted) {
+        setColumns(Loaded(c));
+      }
+    });
+    return () => {
+      mounted = false;
+    };
```

Comment:
question: what is the benefit of all this business with `mounted`? It stops `setColumns` from being called if you navigate away while the api call is happening?

Reply:
Also, I would suggest you extract the api call into a function and use async for readability and error handling. Here's what I have from the column picker ticket in progress, though it could probably still be improved:
```
const fetchColumns = useCallback(async () => {
    try {
      const response = await getProjectColumns({ id: project.id });

      setProjectColumns(Loaded(response));
    } catch (e) {
      handleError(e, { publicSubject: 'Unable to fetch project columns.' });
    }
  }, [project.id]);

  useEffect(() => {
    fetchColumns();
  }, [fetchColumns]);
```

Reply:
re: mounted: `mounted` serves two purposes. One is to prevent the state being set when unmounted as you mentioned, but it also prevents issues with race conditions. If, say, the project id changes without being unmounted  and the columns request of the new project returns before the older one, we want to ensure we don't overwrite the new (faster) state with the older one.

Reply:
`fetchExperiments` isn't using `useEffect`, so we'd need to guard against race conditions differently.